test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTestUtils/Utils.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Tests.HubEndpointTestUtils
+{
+    public class HubEndPointTestUtils
+    {
+        public static Type GetEndPointType(Type hubType)
```

Comment:
Is this only used in `IServiceProvider.GetService`? Might be worth a quick refactor to have this actually be a `GetHubEndpoint` helper that takes an `IServiceProvider` and does all the work in one call.